src/modules/dashboard/components/Dialogs/ControlSafeDialog/ControlSafeDialog.tsx

```diff
@@ -330,6 +338,43 @@ const ControlSafeDialog = ({
     ),
   });
 
+  const transform = useCallback(
+    pipe(
+      mapPayload(
+        ({
+          safe,
+          transactionsTitle,
+          transactions,
+          annotation: annotationMessage,
+        }) => {
+          const chainName = getChainNameFromSafe(safe);
+          const transformedSafe: ColonySafe = {
+            // Find will return because input comes from Safe Networks
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            chainId: SAFE_NETWORKS.find(
+              (network) => network.name === chainName,
+            )!.chainId.toString(),
+            contractAddress: safe.id,
+            safeName: safe.profile.displayName.substring(
+              0,
+              safe.profile.displayName.length - (chainName.length + 3),
+            ),
+          };
```

Comment:
@ArmandoGraterol We would also need to store the chainId (or some reference to the chain) since the same contract address could be present on more than one chain. And since this data doesn't get stored (cleanly) when using the SafePicker, we'd still need to do some transformation somewhere. This seems to me to be the place to do it. 

What you think?

Reply:
@ArmandoGraterol Any thoughts before I merge? I have updated to include the module address so it can be used in the saga.